src/lib.rs

```diff
@@ -1,135 +1,97 @@
-mod visitor;
-
-use std::slice;
-
-use serde::{
-    Deserialize, Serialize,
-    ser::{Serializer, SerializeStruct},
-    de::Deserializer,
-};
-
-use visitor::TaggedVisitor;
-
-pub fn tagged_de<'de, T, D>(des: D, ty: &'static str, tag: &'static &'static str) -> Result<T, D::Error>
-where
-    T: Deserialize<'de>,
-    D: Deserializer<'de>,
-{
-    des.deserialize_struct(ty, slice::from_ref(&tag), TaggedVisitor::new(tag))
-}
-
-pub fn tagged_ser<T, S>(val: &T, ser: S, ty: &'static str, tag: &'static str) -> Result<S::Ok, S::Error>
-where
-    T: Serialize,
-    S: Serializer,
-{
-    let mut sv = ser.serialize_struct(ty, 1)?;
-    sv.serialize_field(tag, val)?;
-    sv.end()
-}
-
-#[cfg(test)]
-mod tests {
-    use serde::{Deserialize, Serialize};
-
-    #[derive(Debug, PartialEq, Deserialize, Serialize)]
-    #[serde(untagged)]
-    enum Enum {
-        #[serde(with = "tagged_i32")]
-        TaggedI32(i32),
-        #[serde(with = "tagged_string")]
-        TaggedString(String),
-        Untagged {
-            x: i32,
-            y: i32
-        }
-    }
-
-    mod tagged_i32 {
-        use serde::{Deserializer, Serializer};
-
-        use crate::{tagged_ser, tagged_de};
-
-        pub fn deserialize<'de, D>(deserializer: D) -> Result<i32, D::Error>
-            where
-                D: Deserializer<'de>
-        {
-            tagged_de(deserializer, "TaggedI32", &"tagged_i32")
-        }
-
-        pub fn serialize<S>(val: &i32, serializer: S) -> Result<S::Ok, S::Error>
-            where
-                S: Serializer
-        {
-            tagged_ser(val, serializer, "TaggedI32", "tagged_i32")
+extern crate proc_macro;
+
+use quote::{format_ident, quote, ToTokens};
+use syn::{parse_macro_input, parse_quote, ItemEnum};
+
+use proc_macro::TokenStream;
+use proc_macro2;
+
+#[proc_macro_attribute]
+pub fn mixed_tags(_attrs: TokenStream, item: TokenStream) -> TokenStream {
+    let mut item = parse_macro_input!(item as ItemEnum);
+    let mut result = proc_macro2::TokenStream::new();
+    let enum_ident = item.ident.to_string();
+
+    let tagged_filter = |attr: &syn::Attribute|
+        /*attr.style == syn::AttrStyle::Outer &&*/
```

Comment:
Mb delete this comment?

Reply:
This line is commented and not removed, because this code should compile, according to latest docs on syn, while in fact, it doesn't. I think we need this line, because with it we fully identify "target" attribute (`#[tagged]`) (btw just understood that this code accepts `#[tagged(..tokens..)]`, this is a bug). So, in my opinion, we should wait this bug to be fixed and then uncomment the line.